service/lib/dinstaller/dbus/interfaces/dasd.rb

```diff
@@ -0,0 +1,178 @@
+              # https://gist.github.com/ancorgs/390b064373995595a1b1dfb08d00507a#gistcomment-4502266
+              dbus_method(:Format, "in devices:ao, out result:u") do |devs|
+                busy_while { dasd_format(devs) }
+              end
+            end
+          end
+        end
+
+        # Enables the DASDs devices indentified by the given D-Bus paths
+        #
+        # See {#dasds_dbus_method} for information about the possible return codes.
+        #
+        # Succeeds (returns 0) if all the devices where successfully enabled.
+        #
+        # @param paths [Array<String>]
+        # @return [Integer]
+        def dasd_enable(paths)
+          dasds_dbus_method(paths) { |dasds| dasd_backend.enable(dasds) }
+        end
+
+        # Disables the DASDs devices indentified by the given D-Bus paths
+        #
+        # See {#dasds_dbus_method} for information about the possible return codes.
+        #
+        # Succeeds (returns 0) if all the devices where successfully disabled.
+        #
+        # @param paths [Array<String>]
+        # @return [Integer]
+        def dasd_disable(paths)
+          dasds_dbus_method(paths) { |dasds| dasd_backend.disable(dasds) }
+        end
+
+        # Sets use_diag to the given value for the DASDs devices indentified by the given
+        # D-Bus paths
+        #
+        # See {#dasds_dbus_method} for information about the possible return codes.
+        #
+        # Succeeds (returns 0) if the desired value of use_diag is properly set for all the
+        # given devices.
+        #
+        # @param paths [Array<String>]
+        # @param diag [Boolean] value of the use_diag flag
+        # @return [Integer]
+        def dasd_set_diag(paths, diag)
+          dasds_dbus_method(paths) { |dasds| dasd_backend.set_diag(dasds, diag) }
+        end
+
+        # Format the DASDs devices indentified by the given D-Bus paths
+        #
+        # See {#dasds_dbus_method} for information about the possible return codes.
+        #
+        # Succeeds (returns 0) if all the devices where successfully formatted.
+        #
+        # @param paths [Array<String>]
+        # @return [Integer]
+        def dasd_format(paths)
+          dasds_dbus_method(paths) { |dasds| dasd_backend.format(dasds) }
+        end
+
+        # Finds the DASDs matching the given D-Bus paths and calls the given block on them
+        #
+        # It only tries to execute the given block if all the paths are found. Otherwise it
+        # returns 1 without trying to process any of the DASDs.
+        #
+        # If all the paths are on the D-Bus tree, it runs the block and returns 0 if it succeeds
+        # or 2 otherwise.
+        #
+        # @return [Integer]
+        def dasds_dbus_method(paths)
+          dbus_dasds = dasds_tree.find_paths(paths)
+          if dbus_dasds.size != paths.size
+            logger.error "Unknown path(s) at #{paths}"
```

Comment:
Does it log unknown path somewhere in y2log or not?

Reply:
Now (in the newly added commit) it does ;-)